feed/views.py

```diff
@@ -19,4 +20,10 @@ def get_context_data(self, **kwargs):
        context = super().get_context_data(**kwargs)
        context['form'] = CommentForm()
        context['advert'] = Advert.objects.first()
-       return context 
\ No newline at end of file
+       return context 
+    def dispatch(self, request, *args, **kwargs):
+     blog_post = get_object()  
+    if request.user.is_anonymous and blog_post.membership_status == "pre": self.user.username
+        return redirect('/accounts/login/')  
+        return super().dispatch(request, *args, **kwargs)
```

Comment:
Indenting my friend. This if statement won't be executed inside of the dispatch() method unless its properly indented.